### ISSUE TITLE
SIGSEGV Crash

### ISSUE THREAD

**Issue by shufps** [iotaledger/iota-core#785]
When we restarted some nodes it happened that other nodes crashed with SIGSEGV.

```
2024-02-26T18:48:35Z	ERROR  	Protocol                           	network error (peer=12D3KooWLQuTQLHS3qNf4obHSj1g8bCfiToW4HgpxbtjbNsBqTZz error=failed to deserialize block: failed to retrieve API for version 3: no engine instance available)

...repeats a couple of dozen of times ...

2024-02-26T18:48:35Z	ERROR  	Protocol                           	network error (peer=12D3KooWLQuTQLHS3qNf4obHSj1g8bCfiToW4HgpxbtjbNsBqTZz error=failed to deserialize block: failed to retrieve API for version 3: no engine instance available)
2024-02-26T18:48:35Z	ERROR  	Protocol                           	network error (peer=12D3KooWLQuTQLHS3qNf4obHSj1g8bCfiToW4HgpxbtjbNsBqTZz error=failed to deserialize block: failed to retrieve API for version 3: no engine instance available)
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x160 pc=0xdba5c2]
```
Here is a bigger chunk of the log:

[log.txt.zip](https://github.com/iotaledger/iota-core/files/14445131/log.txt.zip)

**Comment by alexsporn** [comment]
@jonastheis this might be related to your PR #780

**Comment by jonastheis** [comment]
The logs seem to be truncated but as far as I can tell, it looks like it's the same as we're seeing in #757 and it should indeed be fixed with #780. Due to the restarts the chain probably halted and was revived by several nodes. Due to the bug a node that doesn't itself produce commitments switches chains basically immediately, as can be seen below.

Specifically, this part of the log gives it away: the node sees a new `HeaviestClaimedCandidate` chain and pretty much immediately sets it as `HeaviestAttestedCandidate` and `HeaviestVerifiedCandidate`. For a chain to become  `HeaviestAttestedCandidate`, a node needs to download and verify all attestations. For a chain to become `HeaviestVerifiedCandidate`, a node needs to instantiate a candidate engine, and produce the commitments itself. None of this happens here.

```
2024-02-26T18:48:34Z	TRACE  	Protocol.Chains                    	HeaviestClaimedCandidate (set=1)
[...]
2024-02-26T18:48:34Z	TRACE  	Protocol.Chains.1                  	RequestAttestations (set=true)
2024-02-26T18:48:35Z	TRACE  	Protocol.Commitments.Slot31338.1   	RequestAttestations (set=true)
2024-02-26T18:48:35Z	DEBUG  	Protocol.Attestations              	request (commitment=Slot31338.1)
2024-02-26T18:48:35Z	TRACE  	Protocol.Chains                    	HeaviestAttestedCandidate (set=1)
2024-02-26T18:48:35Z	DEBUG  	Protocol.Chains.1                  	StartEngine (set=true)
2024-02-26T18:48:35Z	TRACE  	Protocol.Commitments.Slot31338.1   	RequestAttestations (set=false)
2024-02-26T18:48:35Z	TRACE  	Protocol.Chains                    	HeaviestVerifiedCandidate (set=1)
2024-02-26T18:48:35Z	TRACE  	Protocol.Chains                    	Main (set=1)
```
